### PR TITLE
My all Task

### PR DESCRIPTION
**Pull request by VasiliZas**
My all Task. I had some problems with IDea and branches, I had to make a new branch with all the tasks.

### CODE REVIEW

**Review by PavelBoPavel — APPROVED** [review]
+

**Review by PavelBoPavel — APPROVED** [review]
+

**Comment by KaydunovDenis** [comment]
Ask Mask your qustion on lecture.

File: homework/Vasili_Zasinets/hw0713/src/main/java/com/tutrit/java/quickstart/homework/Vasili_Zasinets/hw0713/Butterfly.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+                {0, 0, 0, 1, 0, 0, 0}, {0, 0, 1, 1, 1, 0, 0}, {0, 1, 1, 1, 1, 1, 0}, {1, 1, 1, 1, 1, 1, 1}};
+
+
+        for (int k = 0; k < butterfly.length; k++) {
+           // for (int i = 0; i < butterfly.length; i++) {
+               log.info("butterfly: {} " + butterfly[k][0] + butterfly[k][1]+ butterfly[k][2]+ butterfly[k][3]+
+                       butterfly[k][4] + butterfly[k][5]+ butterfly[k][6]);
+            }
+            System.out.println();
+        }
```
Comment (max-java) [inline homework/Vasili_Zasinets/hw0713/src/main/java/com/tutrit/java/quickstart/homework/Vasili_Zasinets/hw0713/Butterfly.java:21]: main method is neither for demo nor for tests. Please, move this code where it should be

File: homework/Vasili_Zasinets/hw0713/src/main/java/com/tutrit/java/quickstart/homework/Vasili_Zasinets/hw0713/LookupArrayService.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LookupArrayService {
+    static Logger log = LoggerFactory.getLogger("Array");
+    int arrayMin = 0;
+    int arrayMax = 0;
+    int indexMin = -1;
+    int indexMax = -1;
```
Comment (max-java) [inline homework/Vasili_Zasinets/hw0713/src/main/java/com/tutrit/java/quickstart/homework/Vasili_Zasinets/hw0713/LookupArrayService.java:11]: I guess logic should works that way in witch these vars are redundant.

File: homework/Vasili_Zasinets/hw0713/src/main/java/com/tutrit/java/quickstart/homework/Vasili_Zasinets/hw0713/LookupArrayService.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+    public int findMax(int[] array) {
+        if (array == null) return 0;
+        arrayMax = array[0];
```
Comment (max-java) [inline homework/Vasili_Zasinets/hw0713/src/main/java/com/tutrit/java/quickstart/homework/Vasili_Zasinets/hw0713/LookupArrayService.java:15]: yep, you could just have 
```
int arrayMax = array[0];
```

File: homework/Vasili_Zasinets/hw0713/src/main/java/com/tutrit/java/quickstart/homework/Vasili_Zasinets/hw0713/NumberService.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
+package com.tutrit.java.quickstart.homework.Vasili_Zasinets.hw0713;
+
+public class NumberService {
+    int result;
+    int i;
+
+
+    int summForSmall(int start, int finish) {
```
Comment (max-java) [inline homework/Vasili_Zasinets/hw0713/src/main/java/com/tutrit/java/quickstart/homework/Vasili_Zasinets/hw0713/NumberService.java:8]: I don't get it, what do you mean by sum for small?

File: homework/Vasili_Zasinets/hw0713/src/main/java/com/tutrit/java/quickstart/homework/Vasili_Zasinets/hw0713/NumberService.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        if (start > finish) {
+            summForBig(start, finish);
+        }
+        else{
+            summForSmall(start, finish);
+        }
+        return result;
+    }
+
+    int selectionForEven(int start, int finish) {
```
Comment (max-java) [inline homework/Vasili_Zasinets/hw0713/src/main/java/com/tutrit/java/quickstart/homework/Vasili_Zasinets/hw0713/NumberService.java:36]: Please, consider self-explanatory names for methods.

File: homework/Vasili_Zasinets/hw0718/src/main/java/com/tutrit/java/quickstart/homework/Vasili_Zasinets/hw0718/ArrayService.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        for (int i = 0; i < array.length; i++) {
+            summOfArray += array[i];
+        }
+        return summOfArray;
+    }
+
+    public double avg(int[] array) {
+        if (array == null) {
+            return 0;
+        }
```
Comment (max-java) [inline homework/Vasili_Zasinets/hw0718/src/main/java/com/tutrit/java/quickstart/homework/Vasili_Zasinets/hw0718/ArrayService.java:38]: this verification could be encapsulated in separate method

File: homework/Vasili_Zasinets/hw0718/src/test/java/com/tutrit/java/quickstart/homework/Vasili_Zasinets/hw0718/ArrayServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    public void setUp() {
+        this.array = arrayService.create(10);
+    }
+
+    @Test
+    public void create() {
+        for (int i = 0; i < array.length; i++) {
+            log.info("array {} ", i + " = " + array[i]);
+        }
+
```
Comment (max-java) [inline homework/Vasili_Zasinets/hw0718/src/test/java/com/tutrit/java/quickstart/homework/Vasili_Zasinets/hw0718/ArrayServiceTest.java:24]: I don't see any service calls here... Should you test it?

File: homework/Vasili_Zasinets/hw0713/src/test/java/com/tutrit/java/quickstart/hw0713/LookupArrayServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+public class LookupArrayServiceTest {
+
+    LookupArrayService lookupArrayService = new LookupArrayService();
+    int[] array;
+    int max;
+    int min;
+    int indexMin;
+    int indexMax;
+    int i;
```
Comment (max-java) [inline homework/Vasili_Zasinets/hw0713/src/test/java/com/tutrit/java/quickstart/hw0713/LookupArrayServiceTest.java:16]: this don't belong here. It's could be move in tests according it's purposes.

**Review by max-java — CHANGES_REQUESTED** [review]
In general looks good, I've got minors comments according to production best practices. 
Nevertheless, please:
1. use quickstart project for every task
2. run sonarqube verification
3. make one mergeRequest per task

**Pull request closed without merging**